### PR TITLE
Added slides on godbolt and cppinsights

### LINKED ISSUE #70: mention cppinsights when talking of e.g. lambdas
it's useful to know what the compiler generates from a lambda function and there is a great website to look at that: https://cppinsights.io/lnk?code=I2luY2x1ZGUgPHZlY3Rvcj4KI2luY2x1ZGUgPGFsZ29yaXRobT4KCmludCBtYWluKCkKewogIHN0ZDo6dmVjdG9yPGludD4gdnsxLCAyLCAzfTsKICBpbnQgaW5jcmVtZW50ID0gNTsKICBzdGQ6OnRyYW5zZm9ybShiZWdpbih2KSwgZW5kKHYpLCBiZWdpbih2KSwKICAgIFtpbmNyZW1lbnRdKGludCB2YWx1ZSkgewogICAgICByZXR1cm4gdmFsdWUgKyBpbmNyZW1lbnQ7CiAgICB9KTsKfQ==&insightsOptions=cpp17&std=cpp17&rev=1.0
it's not just handy for lambdas, you can also check what the compilers creates for e.g. range-for

### PR DESCRIPTION
**Pull request by sponce**
closes #70 and #40

### CODE REVIEW

File: talk/tools/webtools.tex [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    \item range based loops
+    \item templates
+    \item initializations
+    \item auto
+    \item ...
+    \end{itemize}
+  \end{block}
+  \begin{exampleblock}{Typical usage}
+    \begin{itemize}
+    \item understand how things are in the back of \cpp syntax
```
Comment (bernhardmgruber) [inline talk/tools/webtools.tex:54]:
```suggestion
    \item understand how things work behind the \cpp syntax
```

File: talk/tools/webtools.tex [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    \item debug some non working pieces of code
+    \end{itemize}
+  \end{exampleblock}
+\end{frame}
+
+\begin{frame}
+  \frametitle{cppinsights by example}
+  \begin{block}{Check how range based loop work}
```
Comment (bernhardmgruber) [inline talk/tools/webtools.tex:62]:
```suggestion
  \begin{block}{Check how range-based loops work}
```

File: talk/tools/webtools.tex [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    \url{https://cppinsights.io/s/b886aa76}
+    \begin{itemize}
+    \item See how they map to regular iterator
```
Comment (bernhardmgruber) [inline talk/tools/webtools.tex:65]:
```suggestion
    \item See how they map to regular iterators
```

File: talk/tools/webtools.tex [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  \begin{block}{Concept}
+    An online generic compiler with immediate feedback.
+    Allows to :
+    \begin{itemize}
+    \item compile online any code against any version of any compiler
+    \item inspect the assembly generated
+    \item use external libraries (over 50 available !)
+    \item run the code produced
+    \item use tools, e.g. ldd, include-what-you-use, ...
+    \item share small pieces of code via permanent short links
```
Comment (hageboeck) [inline talk/tools/webtools.tex:14]: I wouldn't use "Allow to", see e.g. [english stackexchange](https://english.stackexchange.com/questions/60271/grammatical-complements-for-allow/60285#60285).

We could use:
```suggestion
    Allows:
    \begin{itemize}
    \item compiling any code online against any version of any compiler
    \item inspecting the assembly generated
    \item use of external libraries (over 50 available !)
    \item running the code produced
    \item using tools, e.g.\ ldd, include-what-you-use, ...
    \item sharing small pieces of code via permanent short links
```

**Comment by sponce** [comment]
I've now implemented all comments

**Comment by sponce** [comment]
And switched to pngs, you're absolutely right @bernhardmgruber !

**Review by bernhardmgruber — APPROVED** [review]

File: talk/tools/webtools.tex [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    \item use of external libraries (over 50 available !)
+    \item running the code produced
+    \item using tools, e.g. ldd, include-what-you-use, ...
+    \item sharing small pieces of code via permanent short links
+    \end{itemize}
+  \end{block}
+  \begin{exampleblock}{Typical usage}
+    \begin{itemize}
+    \item check small pieces of code on different compilers
+    \item check some new \cpp functionnality and its support
```
Comment (bernhardmgruber) [inline talk/tools/webtools.tex:20]:
```suggestion
    \item check some new \cpp functionality and its support
```

Comment (sponce) [reply talk/tools/webtools.tex]: done

**Pull request merged**